FF1Lib/Hacks.cs

```diff
@@ -444,19 +444,19 @@ public void EnableDash()
 			Put(0x7D077, Blob.FromHex("A5424A69004A69000A242050014A853460"));
 		}
 
+        /* This replaces the BuyTen function with Disch's item shop enhancement mod */
 		public void EnableBuyTen()
 		{
-			Put(0x380FF, Blob.FromHex("100001110001120001130000"));
-			Put(0x38248, Blob.FromHex("8BB8BC018BB8BCFF8180018EBB5B00"));
-			Put(0x3A8E4, Blob.FromHex("A903203BAAA9122026AAA90485634C07A9A903203BAAA91F2026AAA90385634C07A9EA"));
-			Put(0x3A32C, Blob.FromHex("71A471A4"));
-			Put(0x3A45A, Blob.FromHex("2066A420EADD20EFA74CB9A3A202BD0D039510CA10F860A909D002A925205BAAA5664A9009EAEAEAEAEAEAEAEAEA20F5A8B0E3A562F0054A90DCA909856AA90D205BAA2057A8B0D82076AAA66AF017861318A200A003BD0D0375109D0D03E888D0F4C613D0EB2068AA20C2A8B0ADA562D0A9208CAA9005A9104C77A4AE0C03BD206038656AC9649005A90C4C77A49D206020F3A4A9134C77A4A200A00338BD1C60FD0D039D1C60E888D0F34CEFA7"));
-			Put(0x3AA65, Blob.FromHex("2076AAA90E205BAA2066A4208E8E4C32AAA662BD00038D0C0320B9ECA202B5109D0D03CA10F860A202BD0D03DD1C60D004CA10F51860"));
-			Put(0x3A390, Blob.FromHex("208CAA"));
-			Put(0x3A3E0, Blob.FromHex("208CAA"));
-			Put(0x3A39D, Blob.FromHex("20F3A4"));
-			Put(0x3A404, Blob.FromHex("20F3A4"));
-			Put(0x3AACB, Blob.FromHex("18A202B5106A95109D0D03CA10F5"));
+			Put(0x3801A, Blob.FromHex("B380C480CE80D580EA8052800681198134813F81498160816C819281B681D781F7811C8243824B8261827B829D82B882C2"));
+            /* 8489 = "49", 8383 = "33", 040A21 = quantities to buy */
+			Put(0x380B0, Blob.FromHex("5BC000A0ABA421A7B205BCB2B805BAA4B1B7C5000190B2AFA7019894C500A2A8B60197B200A2B2B805A6A4B1BEB705A4A9A935A705B7AB39C000A0ABB205BAAC4E05B7A4AEA805ACB7C50010000511000512000513009DAB22AE05BCB2B8C405A0AB3905A8AF3EC500A0ABB23E05ACB7A8B005A7B2502605BAA4B1B71BB205B6A84EC500A0ABACA6AB05B2B1A8C5009DAB22AE05BCB2B8C400A0ABB205BAAC4E05AFA82FB105B7ABA805B6B3A84EC500A0ABACA6AB05B6B3A84EC5009CB2B5B5BCBF05A2B2B805A6A4B1BEB705AFA82FB105B7AB39C0059CB2343CA805A8AF3EC500A2B2B805A4AF23A4A7BC05AEB14605B7AB3905B6B3A84EC0059CB2343CA805A8AF3EC500A0A8AFA6B23405FFFF69059CB7A4BCBF05B7B224A43205BCB25505A7A4B7A4C0008DB2B1BEB705A9B2B566B7BF05ACA9502605AFA8A43205BCB25505AAA434BF0091B2AFA7059B8E9C8E9D05BAABAC4505BCB2B805B7B8B5B1059998A08E9B05B2A9A9C4C400A2B2B805ABA43205B1B21C1FAA05B7B224A84E05FFFF69058AB1BCC205B7AB1FAA05A8AF3EC500914605B022BCC500A0ABB205B6ABA44E05A5A805B5A8B9AC32A705C3C300A08A9B9B92989B05C3C3059BA8B755B105B7B205AFACA9A8C4009DAB3005AFA832AF05B6B3A84E05ACB643B84E05C3C3059CB2343CA805A8AF3EC500A2B264A7B205B1B2B705B1A8A827B0BC05ABA8AFB305B1B2BAC000810184018180018383009DB22EA5A4A7056969059C49A8C2051C1FAA05A8AF3EC5000000000000000000000001040A218D0E03A5118D0F0328D001604C32AA688D1D60688D1E6060"));

```

Comment:
I'm not sure how I would even do that.  The code is based on an IPS file that was created by Disch.

Reply:
https://www.masswerk.at/6502/disassembler.html is what I generally use, but if this doesnt come with some commented assembly, its not really important since disassembling it whenever we need to mess with this is not much trouble.

Reply:
Unfortunately there is no ASM file provided with the patch.

Reply:
I can look at translating the code to readable assembly language later tonight.
Does the patch work as advertised?

Reply:
@one1fineday Yes, the patch works and I've used it several times in my custom build of the randomizer.

Reply:
I'll do it tomorrow.